widget_driver_generator/lib/src/utils/default_return_value_helper.dart

```diff
@@ -0,0 +1,106 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+/// Class that provides logic to determine default return values.
+class DefaultReturnValueHelper {
+  /// Return the default value written as code inside a String for the passed in [type].
+  /// It is important to call [hasDefaultValueForType] first to make sure a default value exists.
+  /// Otherwise the code generation might generate faulty code.
+  static String getDefaultValueFor(DartType type) {
+    if (type.element is EnumElement) {
+      // We can safely take the first enum value since dart enforces for enums to have at least one value.
+      return "${_getSanitizedTypeName(type.toString())}.values[0]";
+    }
+    if (type is RecordType) {
+      return _getDefaultValueForRecord(type);
+    }
+    if (type is InterfaceType && type.element is ClassElement && type.element.name == "Future") {
+      final defaultValue = getDefaultValueFor(type.typeArguments.first);
+      return "Future.value($defaultValue)";
+    } else {
+      return _getDefaultValueForTypeName(type.toString());
+    }
+  }
+
+  /// Returns `true` if a default value exists for the passed in [type].
+  /// If it returns `false`, [getDefaultValueFor] might not return a proper value and should not be used.
+  static bool hasDefaultValueForType(DartType type) {
+    if (type.element is EnumElement) {
+      return true;
+    }
+    // RecordTypes are tuples (value1, value2) that were introduced with dart 3.0 and are a edge case here
+    if (type is RecordType) {
+      return _hasDefaultValueForRecord(type);
+    }
+    var typeName = type.toString();
+    if (type is InterfaceType && type.element is ClassElement && type.element.name == "Future") {
+      typeName = type.typeArguments.first.toString();
+    }
+    return _hasDefaultValueForTypeName(typeName);
+  }
+
+  // to introduce more types with default values, add them to this map.
+  // All enum values and optional values are already covered and don't have to be put in here.
+  static final Map<String, String> _defaultTypeValueMap = {
+    "int": "0",
+    "double": "0.0",
+    "String": "''",
```

Comment:
Should maybe the default string value contain a space? Like `"String": "' '",`
I can imagine that it is often used in some text-widget.. And maybe they render strange if they have an empty string.
What do you think? I can guess that it could give you some funky widget-test-errors sometimes.

Reply:
I guess there is no harm in adding the whitespace.